mingw-w64-git/git-wrapper.c

```diff
@@ -113,15 +116,52 @@ static void my_path_append(LPWSTR list, LPCWSTR path, size_t alloc)
 	}
 }
 
+bool IsRunningOnARM64()
+{
+	USHORT processMachine = 0;
+	USHORT nativeMachine = 0;
+
+	// Note: IsWow64Process2 is only available in Windows 10 1511+
+	LPFN_ISWOW64PROCESS2 fnIsWow64Process2 = (LPFN_ISWOW64PROCESS2)GetProcAddress(
+		GetModuleHandle(L"kernel32"), "IsWow64Process2");
```

Comment:
I _slightly_ prefer doing this without that `typedef`'ed `LPFN_*`:

```c
BOOL (WINAPI* IsWow64Process2)(HANDLE, PUSHORT, PUSHORT) =
    GetProcAddress(GetModuleHandle(L"kernel32"), "IsWow64Process2");

return IsWow64Process2 &&
    IsWow64Process(GetCurrentProcess(), &process_machine, &native_machine) &&
    native_machine == 0xaa64;
```

Reply:
@dscho thanks for the suggestion. I tried implementing it, but it leads to compiler errors:

```
../git-wrapper.c: In function 'IsRunningOnARM64':
../git-wrapper.c:129:39: warning: passing argument 2 of 'IsWow64Process' from incompatible pointer type [-Wincompatible-pointer-types]
  129 |   IsWow64Process(GetCurrentProcess(), &process_machine, &native_machine) &&
      |                                       ^~~~~~~~~~~~~~~~
      |                                       |
      |                                       USHORT * {aka short unsigned int *}
In file included from C:/git-sdk-64/mingw32/i686-w64-mingw32/include/winbase.h:41,
                 from C:/git-sdk-64/mingw32/i686-w64-mingw32/include/windows.h:70,
                 from ../git-wrapper.c:11:
C:/git-sdk-64/mingw32/i686-w64-mingw32/include/wow64apiset.h:23:68: note: expected 'PBOOL' {aka 'int *'} but argument is of type 'USHORT *' {aka 'short unsigned int *'}
   23 |   WINBASEAPI WINBOOL WINAPI IsWow64Process (HANDLE hProcess, PBOOL Wow64Process);
      |                                                              ~~~~~~^~~~~~~~~~~~
../git-wrapper.c:129:3: error: too many arguments to function 'IsWow64Process'
  129 |   IsWow64Process(GetCurrentProcess(), &process_machine, &native_machine) &&
      |   ^~~~~~~~~~~~~~
In file included from C:/git-sdk-64/mingw32/i686-w64-mingw32/include/winbase.h:41,
                 from C:/git-sdk-64/mingw32/i686-w64-mingw32/include/windows.h:70,
                 from ../git-wrapper.c:11:
C:/git-sdk-64/mingw32/i686-w64-mingw32/include/wow64apiset.h:23:29: note: declared here
   23 |   WINBASEAPI WINBOOL WINAPI IsWow64Process (HANDLE hProcess, PBOOL Wow64Process);
      |                             ^~~~~~~~~~~~~~
make: *** [../mingw-w64-git.mak:8: git-wrapper.o] Error 1
```

I do have a feeling that the `typedef`'ed `LPFN_*` is there for a reason, but there might still be alternatives that are cleaner. Do you have any other suggestions?

Reply:
Ah, that's because I forgot the `2` in the actual call to `IsWow64Process2()` ;-) This should work:

```c
bool IsRunningOnARM64()
{
	USHORT process_machine = 0;
	USHORT native_machine = 0;

	// Note: IsWow64Process2 is only available in Windows 10 1511+
	BOOL (WINAPI* IsWow64Process2)(HANDLE, PUSHORT, PUSHORT) =
    GetProcAddress(GetModuleHandle(L"kernel32"), "IsWow64Process2");

	return IsWow64Process2 &&
		IsWow64Process2(GetCurrentProcess(), &process_machine, &native_machine) &&
		native_machine == 0xaa64;
}
```

Reply:
Haha, oops! Completely overlooked that part yesterday. Fixed and works correctly 😄